ast/parser_test.go

```diff
@@ -2296,6 +2296,94 @@ func TestParserText(t *testing.T) {
 	}
 }
 
+func TestRuleText(t *testing.T) {
+	input := `
+r1 = x {
+	a == "foo"
+} else = y {
+	b == "bar"
+}
+
+else {
+	c == "baz"
+}
+
+else = {
+	"k1": 1,
+	"k2": 2
+} {
+	true
+}
+`
+
+	rule := MustParseRule(input)
+	assertLocationText(t, strings.TrimSpace(input), rule.Location)
+	assertLocationText(t, "r1 = x", rule.Head.Location)
+	assertLocationText(t, "x", rule.Head.Value.Location)
+
+	curElse := rule.Else
+	if curElse == nil {
+		t.Fatalf("Expected an else block, got nil")
+	}
+	assertLocationText(t, strings.TrimSpace(`
```

Comment:
What does the old parser do for else text? I would have expected every else block to have the text for the else blocks underneath. The else _head_ text looks great.

Reply:
I'll double check what it does. I guess either way I would agree that it makes sense for the else to have all the subsequent ones if thats what we do for the top level rule.